src/sql/azdata.proposed.d.ts

```diff
@@ -1790,6 +1795,77 @@ declare module 'azdata' {
 		 * The path of the parent node.
 		 */
 		parentNodePath: string;
+		/**
+		 * flag to indicate if the node supports filtering
+		 */
+		isFilteringSupported?: boolean;
+		/**
+		 * list of filters that are supported by the node
+		 */
+		defaultFilters?: NodeInfoFilterDefinition[];
+		/**
+		 * list of properties to filter children nodes on
+		 */
+		filters?: NodeInfoFilter[];
+	}
+
+	export interface NodeInfoFilterDefinition {
+		/**
+		 * The name of the filter property
+		 */
+		name: string;
+		/**
+		 * The type of the filter property
+		 */
+		type: NodeInfoFilterDataType;
+		/**
+		 * The list of options for the filter property if the type is enum
+		 */
+		options?: string[];
+		/**
+		 * The description of the filter property
+		 */
+		description: string;
+	}
+
+	export interface NodeInfoFilter {
+		/**
+		 * The name of the filter property
+		 */
+		name: string;
+		/**
+		 * The type of the filter property
+		 */
+		type: NodeInfoFilterDataType;
+		/**
+		 * The operator of the filter property
+		 */
+		operator?: NodeInfoFilterOperators;
+		/**
+		 * The applied values of the filter property
+		 */
+		values: string[];
+	}
+
+	export enum NodeInfoFilterDataType {
+		string = 0,
+		number = 1,
+		boolean = 2,
+		date = 3,
+		enum = 4
```

Comment:
What about "choice" or "array" instead of enum? The values don't necessarily have to be enums, and they aren't usually going to behave in the same way.